### PR TITLE
Add option to clear command cooldowns on online event

### PR DESCRIPTION
**Pull request by Sartharon**
- Adds an option to clear active cooldowns on online event (Allows users to create "single use" commands which can only be used once per stream, even if they stream multiple times a day)
- Adds a command that clears all active cooldowns
- Adds a command that clears the active cooldowns of a specific command
- Refactors the `handleCoolCom`-function and the output messages to reduce code duplication and readability
- Clear active cooldowns if cooldown time is unset (used to only work by changing the cooldown time via command, via panel works now as well)

image
image

### CODE REVIEW

**Comment by Sartharon** [comment]
Draft:
Cody changes will come tomorrow
Save missing after clearing cooldowns

**Comment by Sartharon** [comment]
Not sure what Codacy means with `Delete ´..´`
Other warnings are about having to use new lines in the lang files ... I could do that for all the lang files if it's something that's wanted

**Comment by gmt2001** [comment]
Indents apparently. I disabled it

**Comment by Sartharon** [comment]
> Indents apparently. I disabled it

I figured as much and redid them for the mentioned lines but no dice

File: javascript-source/core/commandCoolDown.js [hunk 1]
```diff
@@ -310,158 +324,209 @@
...
-            modsSkip = 'ModsSkip';
-        } else {
-            modsSkip = '';
+        $.say($.whisperPrefix(sender) + message);
+    }
+
+    /**
+     * @event twitchOnline
+     */
+    $.bind('twitchOnline', function () {
```
Comment (gmt2001) [inline javascript-source/core/commandCoolDown.js:463]: You should re-read this function

File: javascript-source/core/misc.js [hunk 2]
```diff
@@ -434,6 +434,42 @@
...
+     */
+    function stringToBoolean(string) {
+        if (string === undefined || string === null) {
+            return null;
+        }
+        if (string.equalsIgnoreCase('true') || string === '1') {
+            return true;
+        }
+        if (string.equalsIgnoreCase('false') || string === '0') {
+            return true;
```
Comment (gmt2001) [inline javascript-source/core/misc.js]: Test fail
```js
stringToBoolean('false') === false
```

File: javascript-source/core/misc.js [hunk 3]
```diff
@@ -434,6 +434,42 @@
...
+    /**
+     * Converts any possible boolean represented as a string ('1', '0', 'true', 'false') to a proper boolean
+     * @param {String} string
+     * @return the boolean; null if it's not a valid boolean represented as a string
+     */
+    function stringToBoolean(string) {
+        if (string === undefined || string === null) {
+            return null;
+        }
+        if (string.equalsIgnoreCase('true') || string === '1') {
```
Comment (gmt2001) [inline javascript-source/core/misc.js]: `TypeError: Property 'equalsIgnoreCase' does not exist on object 'someString'`

Use `$.equalsIgnoreCase(str)` instead for a safe implementation

File: javascript-source/core/misc.js [hunk 4]
```diff
@@ -434,6 +434,42 @@
...
+            return true;
+        }
+        if (string.equalsIgnoreCase('false') || string === '0') {
```
Comment (gmt2001) [inline javascript-source/core/misc.js]: `TypeError: Property 'equalsIgnoreCase' does not exist on object 'someString'`

Use `$.equalsIgnoreCase(str)` instead for a safe implementation

**Review by gmt2001 — CHANGES_REQUESTED** [review]
Missing update task to set default `clearOnOnline` on existing cooldowns

File: javascript-source/core/commandCoolDown.js (hunk 1 above)
Comment (Sartharon) [reply javascript-source/core/commandCoolDown.js:463]: God ... was I blind that evening
Thanks for the review

**Comment by Sartharon** [comment]
### [Codacy](https://app.codacy.com/manual/PhantomBot/PhantomBot/commit?cid=742249942) detected an issue:
#### Message: `Generic Object Injection Sink`

    
#### Occurred on:
+ **Commit**:  f087a084f12ac0b6249cba07aa9af7d52d333ec6
+ **File**:    [javascript-source/core/commandCoolDown.js](https://github.com/PhantomBot/PhantomBot/blob/f087a084f12ac0b6249cba07aa9af7d52d333ec6/javascript-source/core/commandCoolDown.js)
+ **LineNum**: [331](https://github.com/PhantomBot/PhantomBot/blob/f087a084f12ac0b6249cba07aa9af7d52d333ec6/javascript-source/core/commandCoolDown.js#L331)
+ **Code**:    `cooldowns[command].globalTime = 0;`

    
#### Currently on:
+ **Commit**:  1bdeead08801c9d855bb5bc0ad0947ead308bff4
+ **File**:    [javascript-source/core/commandCoolDown.js](https://github.com/PhantomBot/PhantomBot/blob/1bdeead08801c9d855bb5bc0ad0947ead308bff4/javascript-source/core/commandCoolDown.js)
+ **LineNum**: [302](https://github.com/PhantomBot/PhantomBot/blob/1bdeead08801c9d855bb5bc0ad0947ead308bff4/javascript-source/core/commandCoolDown.js#L302)
   

Command is validated as best we can (it's not null nor undefined `L#289`). More info on how this would normally be resolved [here](https://stackoverflow.com/questions/51715616/how-to-fix-codacy-alert-generic-object-injection-sink)
Plus this is still in beta as stated by codacy:
> For this tool we are not yet meeting this standard but you can help us improve the docs

**Review by gmt2001 — APPROVED** [review]

**Pull request merged**